test/bats/aws.bats

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env bats
+
+load helpers
+
+
+WAIT_TIME=120
+SLEEP_TIME=1
+PROVIDER_YAML=https://raw.githubusercontent.com/aws/secrets-store-csi-driver-provider-aws/main/deployment/aws-provider-installer.yaml
+NAMESPACE=kube-system
+CLUSTER_NAME=integ-cluster
+POD_NAME=basic-test-mount
+export REGION=us-west-2
+export ACCOUNT_NUMBER=$(aws --region $REGION  sts get-caller-identity --query Account --output text)
+
+BATS_TEST_DIR=test/bats/tests/aws
+
+setup_file() {
+    
+    #Create and initialize cluster 
+
+    eksctl create cluster \
+       --name $CLUSTER_NAME \
+       --node-type m5.large \
+       --nodes 3 \
+       --region $REGION
+
+   eksctl utils associate-iam-oidc-provider --name $CLUSTER_NAME --approve --region $REGION
+    
+   eksctl create iamserviceaccount \
+       --name basic-test-mount-sa \
+       --namespace $NAMESPACE \
+       --cluster $CLUSTER_NAME \
+       --attach-policy-arn arn:aws:iam::aws:policy/AmazonSSMReadOnlyAccess \
+       --attach-policy-arn arn:aws:iam::aws:policy/SecretsManagerReadWrite \
+       --override-existing-serviceaccounts \
+       --approve \
+       --region $REGION
+
+    #Install csi secret driver
+    helm repo add secrets-store-csi-driver https://raw.githubusercontent.com/kubernetes-sigs/secrets-store-csi-driver/master/charts
+    helm --namespace=$NAMESPACE install csi-secrets-store secrets-store-csi-driver/secrets-store-csi-driver --set enableSecretRotation=true --set rotationPollInterval=15s 
```

Comment:
The driver PR tests as running with the image that's built as part of the PR: https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/master/Makefile#L278-L290.

Could we move the cluster creation, image build to a script that's outside of the bats test suite?

Reply:
Yes I can do that and it would be more consistent with the existing workflow

Reply:
I also add a similar cleanup task/target - https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/master/Makefile#L304